### DISCUSSION TITLE
About soft bodies collision detection

### DISCUSSION THREAD

**Discussion by cold-young** [NVIDIA-Omniverse/PhysX#80] [Q&A] [answered]
Hi, 
I want to implement a contact sensor that can detect when a soft object collision.
So, I'd like to determine if the soft body hit each other or not.
I found deformable_body.GetCollisionPointsAttr() but, I don't know how I use this api.
I want to get the mesh information that hit the soft body in real time, but this code is not like that.

Is there any other function that the soft body can detect a contact?

Thanks,

**Comment by preist-nvidia** [comment, marked as answer]
Hi,

The collision points are not providing contact information, but information about the collision tetrahedral mesh vertex positions.

We do not support contact information for soft bodies yet, but this support is on our radar and will be made available in a future release.

**Reply by cold-young** [reply]
Thank you, 
If I want to get contact information about soft bodies, How I can do? 
Actually, this simulation expresses the collision for the soft body. Is there any way I can use what's used here?

**Reply by preist-nvidia** [reply]
You cannot yet get the contact information, this is the support we are planning to add in a future release.

I am not sure I understand your second question, but it would not be straightforward to get more contact information out of the data we do provide.

**Comment by hadar-segal-isi** [comment]
Hi,
I am also very interested in contact information for soft bodies. Any updates on this issue?

**Reply by hadar-segal-isi** [reply]
@preist-nvidia it seems like even scene queries ignore the soft body. Is there some efficient alternative for this? please let me know if this should be a different discussion. Thanks

**Reply by monkeygame** [reply]
Neither simulationEventCallback nor Queries support soft bodies and particle systems, and this is not clearly stated in the documentation. This is very bad. I spent a lot of time figuring out why they didn't work. I always thought it was due to my improper use! @preist-nvidia

**Reply by alpargun** [reply]
Thanks for the information here. So, is SOFA still the way to go?

**Reply by monkeygame** [reply]
I don't think so.
The performance and stability of SOFA are too poor to support large-scale projects.

**Reply by alpargun** [reply]
What simulator do you prefer? We are also working with ANSYS at the moment.
